### PR TITLE
Improve eject media resource allowing force

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref: https://github.com/terraform-providers/terraform-provider-vcd/issues/171

Change which allows to use new property `eject_force`. E.g.
```
resource "vcd_inserted_media" "ISO" {
  count     = "1"
  catalog   = "testCatalog"
  name      = "TestAccVcdCatalogMediaBasicForInsert"
  vapp_name = "vappNetVAppTest"
  vm_name   = "add1"
  eject_force = true
  # depends_on = ["vcd_vapp_vm.vmAmdNet1"]
}
```
This change uses new API introduced in govcd which allows to answer questions which rises when umount happend on running VM. New wrapped ejectTask handle such hurdle.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ ARCHITECTURAL:
 * Project switched to using Go modules, while `vendor` is left for backwards build compatibility only. It is worth having a
 look at [README.md](README.md) to understand how Go modules impact build and development.
 
+IMPROVEMENTS:
+
+* Insert Media now supports force ejecting with running VM
```
Comment (lvirbalas) [inline CHANGELOG.md]: "with running VM" -> "on a running VM"

Overall, appreciate the CHANGELOG addition, thanks.

File: website/docs/r/inserted_media.html.markdown [hunk 2]
```diff
@@ -35,4 +37,5 @@ The following arguments are supported:
 * `catalog` - (Required) The name of the catalog where to find media file
 * `name` - (Required) Media file name in catalog which will be inserted to VM
 * `vapp_name` - (Required) - The name of vApp to find
-* `vm_name` - (Required) - The name of VM to be used to insert media file
\ No newline at end of file
+* `vm_name` - (Required) - The name of VM to be used to insert media file
+* `eject_force` - (Optional; *v2.1+*) Allows to pass answer to question when ejecting with not stopped VM. True means "Yes" as answer to question. Default is `true`
```
Comment (lvirbalas) [inline website/docs/r/inserted_media.html.markdown]: Great to see the *v2.1+* flag.

Comments:
- For the unwary user it's not clear what this "question" is. Need to add a word or two about context.
- "when ejecting with not stopped VM" -> "when ejecting from a VM which is powered on"

File: CHANGELOG.md (hunk 1 above)
Comment (vbauzysvmware) [reply CHANGELOG.md]: Fixed

File: website/docs/r/inserted_media.html.markdown (hunk 2 above)
Comment (vbauzysvmware) [reply website/docs/r/inserted_media.html.markdown]: Added message example

File: website/docs/r/inserted_media.html.markdown [hunk 3]
```diff
@@ -38,4 +38,7 @@ The following arguments are supported:
 * `name` - (Required) Media file name in catalog which will be inserted to VM
 * `vapp_name` - (Required) - The name of vApp to find
 * `vm_name` - (Required) - The name of VM to be used to insert media file
-* `eject_force` - (Optional; *v2.1+*) Allows to pass answer to question when ejecting with not stopped VM. True means "Yes" as answer to question. Default is `true`
\ No newline at end of file
+* `eject_force` - (Optional; *v2.1+*) Allows to pass answer to question
```
Comment (lvirbalas) [inline website/docs/r/inserted_media.html.markdown]: "to question" -> "to question in vCD"

Comment (vbauzysvmware) [reply website/docs/r/inserted_media.html.markdown]: fixed. No failing anymore

File: vcd/resource_vcd_inserted_media_test.go [hunk 4]
```diff
@@ -196,25 +197,27 @@ resource "vcd_vapp_vm" "{{.VmName}}" {
...
-  
-  vapp_name = "${vcd_vapp.{{.VappName}}.name}" 
-  vm_name = "${vcd_vapp_vm.{{.VmName}}.name}" 
+  name    = "{{.CatalogMediaName}}"
+
+  vapp_name  = "${vcd_vapp.{{.VappName}}.name}"
+  vm_name    = "${vcd_vapp_vm.{{.VmName}}.name}"
   depends_on = ["vcd_vapp_vm.{{.VmName}}", "vcd_catalog_media.{{.CatalogMediaName}}"]
+
+  eject_force = "{{.ejectForce}}"
```
Comment (Didainius) [inline vcd/resource_vcd_inserted_media_test.go]: Typo
```suggestion
  eject_force = "{{.EjectForce}}"
```

Currently the test `.tf` file reenders so as it misses the value:
```
$  cat vcd/test-artifacts/vcd.TestAccVcdMediaInsertBasic.tf | tail -5
  vm_name    = "${vcd_vapp_vm.TestAccVcdVAppVmForInsert.name}"
  depends_on = ["vcd_vapp_vm.TestAccVcdVAppVmForInsert", "vcd_catalog_media.TestAccVcdCatalogMediaBasicForInsert"]

  eject_force = ""
}
```

Comment (vbauzysvmware) [reply vcd/resource_vcd_inserted_media_test.go]: Well in sense we don't change default behaviour. Currently with tuning VM you usually stumble on issue.

**Pull request merged**